### PR TITLE
Refactoring Jointpoint utility class to a decorator.

### PR DESCRIPTION
**Pull request by shelan**
This refactored existing JoinPointUtility (removed) class and introduced new ImprovedJoinPoint as a decorator.
issue related to this PR is #168

### CODE REVIEW

**Comment by dmarkov** [comment]
@shelan Thanks, I will find someone to review this PR soon

**Comment by dmarkov** [comment]
@dmzaytsev review this one please

**Comment by dmzaytsev** [comment]
@shelan according our [quality rules ](http://www.teamed.io/qa.html)
> Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

please update the description

File: src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java [hunk 1]
```diff
@@ -32,47 +32,103 @@
...
+ * This class implements the methods from JointPoint interface.
  *
- * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @author Shelan Perera (shelanrc@gmail.com)
  * @version $Id$
- * @since 0.7.22
+ * @since 1.0
  */
-final class JoinPointUtils {
+public class ImprovedJoinPoint implements JoinPoint {
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java]: @shelan you can declare whole class as final

File: src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java [hunk 2]
```diff
@@ -32,47 +32,103 @@
...
+     * Constructor for BaseJointPoint.
+     *
+     * @param jpt Parameter of joinpoint instance for the constructor
+     */
+    ImprovedJoinPoint(final JoinPoint jpt) {
+        this.joinpoint = jpt;
+    }
+
+    @Override
+    public final String toString() {
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java]: @shelan so no need to declare every method as final

File: src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java [hunk 3]
```diff
@@ -32,47 +32,103 @@
...
     /**
-     * Utility class constructor.
+     *
      */
-    private JoinPointUtils() {
-        // intentionally left empty
+    private final transient JoinPoint joinpoint;
+
+    /**
+     * Constructor for BaseJointPoint.
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java]: @shelan why `BaseJointPoint`?
anyway  I think `Constructor.` will be enough or even `Ctor.`

File: src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java [hunk 4]
```diff
@@ -32,47 +32,103 @@
...
+     *
+     * @param jpt Parameter of joinpoint instance for the constructor
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java]: @shelan `* @param jpt Origin instance` will be enough

**Comment by dmzaytsev** [comment]
@shelan just a few minor comments above

**Comment by shelan** [comment]
@dmzaytsev Thank you very much for the review. I have made the suggested changes and updated the PR.

**Comment by dmzaytsev** [comment]
@shelan thanks!

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmarkov** [comment]
@dmzaytsev **19 mins** added to your account (payment number `66774281`), many thanks for your contribution! 96 hours and 10 mins spent here.; you're getting extra minutes here (c=4); +19 added to your rating, current score is: [+1288](http://www.netbout.com/b/35166?open=rating)

**Comment by dmarkov** [comment]
@rultor please deploy

**Pull request merged**